Source/NSButton.m

```diff
@@ -543,4 +547,43 @@ - (NSSound *) sound
   return [_cell sound];
 }
 
+- (BOOL) _isRadioButton
+{
+  return ([[self cell] image] ==
+          [NSImage imageNamed: @"NSRadioButton"]);
+}
+
+- (void) mouseDown: (NSEvent *)event
+{
+  [super mouseDown: event];
+
+  if ([self _isRadioButton] == NO)
+    return;
+  
+  if ([[self superview] isKindOfClass: [NSMatrix class]] == NO)
+  {
+    NSView *sv = [self superview];
+    NSArray *subviews = [sv subviews];
+    
+    FOR_IN(NSView*, v, subviews)
+      {
+        if ([v isKindOfClass: [NSButton class]])
+          {
+            NSButton *b = (NSButton *)v;
+            if ([b _isRadioButton])
+              {
+                if ([self action] == [b action] && b != self)
+                  {
+                    if ([self state] == NSOnState)
```

Comment:
This check is still inside the loop.

Reply:
I have removed the need for this check, so it is gone.